tensorpipe/CMakeLists.txt

```diff
@@ -206,29 +202,56 @@ if(TP_ENABLE_IBV)
     transport/ibv/reactor.cc
     transport/ibv/sockaddr.cc
     transport/ibv/utility.cc)
-  set(TP_PUBLIC_HEADERS ${TP_PUBLIC_HEADERS}
+  list(APPEND TP_PUBLIC_HDRS
     transport/ibv/error.h
     transport/ibv/factory.h
     transport/ibv/utility.h)
   set(TENSORPIPE_HAS_IBV_TRANSPORT 1)
 endif()
 
-if(APPLE)
-  find_library(CF CoreFoundation)
-  find_library(IOKIT IOKit)
-  target_link_libraries(tensorpipe PRIVATE ${CF} ${IOKIT})
+# Handling the file lists
+if(TP_USE_CUDA)
+  list(APPEND TP_PRIVATE_SRCS ${TP_CUDA_PRIVATE_SRCS})
+  list(APPEND TP_PUBLIC_HDRS ${TP_CUDA_HDRS})
 endif()
 
+# Add the library target
+add_library(tensorpipe ${TP_STATIC_OR_SHARED} ${TP_PUBLIC_SRCS})
+target_sources(tensorpipe PRIVATE ${TP_PRIVATE_SRCS})
 
+# Set target properties
 if(BUILD_SHARED_LIBS)
   set_target_properties(tensorpipe PROPERTIES POSITION_INDEPENDENT_CODE 1)
 endif()
 
+# Add uv package
+find_package(uv REQUIRED)
+target_link_libraries(tensorpipe PRIVATE uv::uv)
+
+# If CUDA is required find the CUDA package and add to link deps
+if(TP_USE_CUDA)
+  find_package(CUDA REQUIRED)
+  target_link_libraries(tensorpipe PUBLIC ${CUDA_LIBRARIES})
+  target_include_directories(tensorpipe PUBLIC ${CUDA_INCLUDE_DIRS})
+endif()
+
+# MAC OS specific library deps
+if(APPLE)
+  find_library(CF CoreFoundation)
+  find_library(IOKIT IOKit)
+  target_link_libraries(tensorpipe PRIVATE ${CF} ${IOKIT})
+endif()
```

Comment:
As I said in the other PR, it could be nicer to define some additional variables at the beginning, like `TP_LINK_LIBRARIES` and `TP_INCLUDE_DIRECTORIES`, ..., and then append elements to them as we go (so that we can keep this logic more tightly coupled with the rest and avoid repetition), and at the end just use those variables to populate the `tensorpipe` target.

Reply:
Updated the changes.
Keeping all the dependent libraries link as PRIVATE, so that the interface function dont become public API of tensorpipe library. And keeping the target include directories as PUBLIC, since it is required.